source/weakref/index.rst

```diff
@@ -73,10 +76,9 @@ Référence circulaire
 --------------------
 
 L'utilité des *weakref* n'est pas des plus optimal dans l'exemple présenté ci-dessus. Nous allons donc maintenant rajouter une méthode à notre class.
-Ce qui va nous permettre d'experimenter les références cyclique (c'est un objet qui a dans ces propriétés une référence au même type que lui).
-
-.. Cette explication est confuse.
-
+Ce qui va nous permettre d'expérimenter les références circulaires. Ce type de références arrive quand deux objets se référencent mutuellement,
+ce qui va augmenter leurs compteurs de référence. Et au moment où l'on n'utilise plus ces objets, le ramasse-miettes ne va pas pouvoir supprimer
+les supprimer vu que leur compteur n'est pas à zéro.
```

Comment:
> _While Python uses the traditional reference counting implementation, it also offers a cycle detector that works to detect reference cycles. This allows applications to not worry about creating direct or indirect circular references; these are the weakness of garbage collection implemented using only reference counting. Reference cycles consist of objects which contain (possibly indirect) references to themselves, so that each object in the cycle has a reference count which is non-zero. Typical reference counting implementations are not able to reclaim the memory belonging to any objects in a reference cycle, or referenced from the objects in the cycle, even though there are no further references to the cycle itself._
>
> _The cycle detector is able to detect garbage cycles and can reclaim them. The gc module exposes a way to run the detector (the collect() function), as well as configuration interfaces and the ability to disable the detector at runtime. The cycle detector is considered an optional component; though it is included by default, [...]_
>
> https://docs.python.org/3/extending/extending.html#reference-counts